Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -68,20 +68,45 @@ extension StripeCardReaderService: CardReaderService {
             simulated: true
         )
 
-        // Enough code to pass a test
+        switchStatusToDiscovering()
+
+        /**
+         * https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPTerminal.html#/c:objc(cs)SCPTerminal(im)discoverReaders:delegate:completion:
+         *
+         *Note that if discoverReaders is canceled, the completion block will be called with nil (rather than an SCPErrorCanceled error).
+         */
         discoveryCancellable = Terminal.shared.discoverReaders(config, delegate: self, completion: { error in
-            if let error = error {
-                print("discoverReaders failed: \(error)")
-            } else {
-                print("discoverReaders succeeded")
+            guard let error = error else {
+                self.switchStatusToIdle()
+                return
             }
+
+            self.internalError(error)
```

Comment:
sanity check: Is it ok for us to capture `self` here and in the guard above?

Reply:
I'll weakify self, which, as far as I understand, will make the return in one line mentioned above not possible